features/qskconfig.cmake

```diff
@@ -0,0 +1,185 @@
+############################################################################
+# QSkinny - Copyright (C) 2016 Uwe Rathmann
+# This file may be used under the terms of the QSkinny License, Version 1.0
+############################################################################
+
+set(QSK_VER_MAJ 0)
+set(QSK_VER_MIN 0)
+set(QSK_VER_PAT 0)
+set(QSK_VERSION ${QSK_VER_MAJ}.${QSK_VER_MIN}.${QSK_VER_PAT})
+
+# trying the PREFIX environment variable first
+set(QSK_INSTALL_PREFIX = $ENV{PREFIX}) # TODO still required?
+
+if("${QSK_INSTALL_PREFIX}" STREQUAL "")
+    set(QSK_INSTALL_PREFIX "${CMAKE_INSTALL_PREFIX}") # TODO where does QT_INSTALL_PREFIX come from?
+
+    if (UNIX)
+        set(QSK_INSTALL_PREFIX /usr/local/qskinny-${QSK_VERSION})
+    endif()
+
+    if (MSVC)
+        set(QSK_INSTALL_PREFIX C:/Qskinny-${QSK_VERSION})
+    endif()
+endif()
+
+set(QSK_INSTALL_DOCS      doc)
+set(QSK_INSTALL_HEADERS   include)
+set(QSK_INSTALL_LIBS      lib)
+set(QSK_INSTALL_BINS      bin)
+set(QSK_INSTALL_PLUGINS   plugins)
+set(QSK_INSTALL_EXAMPLES  examples)
+
+# TODO find compiler flag equivalent
+list(APPEND CONFIG        no_private_qt_headers_warning)
+list(APPEND CONFIG        warn_on)
+list(APPEND CONFIG        no_keywords)
+list(APPEND CONFIG        silent)
+#list(APPEND CONFIG           -= depend_includepath) # TODO was -=
+list(APPEND CONFIG        strict_c++)
+list(APPEND CONFIG        c++11)
+list(APPEND CONFIG        pedantic)
+list(APPEND CONFIG        hide_symbols)
+#list(APPEND CONFIG           += debug)
+#list(APPEND CONFIG           += sanitize)
+
+# TODO
+# use_no_rpath {
+#     CONFIG -= use_local_rpath use_install_rpath
+# } else {
+#     cross_compile {
+#         CONFIG *= use_install_rpath
+#         QMAKE_RPATHLINKDIR *= $${QSK_PLUGIN_DIR}/skins
+#     } else {
+#         !use_install_rpath: CONFIG *= use_local_rpath
+#     }
+# }
+
+set(MOC_DIR      moc)
+set(OBJECTS_DIR  obj)
+set(RCC_DIR      rcc)
+
+list(APPEND QSK_CONFIG QskDll)
```

Comment:
@uwerat shall all shared libraries be compilable as static aswell?

Reply:
Actually I never tried to built the libs statically myself, but as we are heading towards a first release under a BSD license in a couple of months it might be of interest for users with a commercial Qt license.

So yes static builds are of interest in general, but it is not mandatory for replacing qmake by cmake.